### PR TITLE
Implement new My Library UI

### PR DESCRIPTION
**Pull request by suecarmol**
[//]: # (Thank you for uploading a PR to the Wikipedia Library!)

## Description
Implement the missing pieces of My Library UI after completing #762. 

## Rationale
[//]: # (Why is this change required? What problem does it solve?)
This is part of TWL redesign efforts.

## Phabricator Ticket
[//]: # (Link to the Phabricator ticket)
[T285556](https://phabricator.wikimedia.org/T285556)

## How Has This Been Tested?
[//]: # (- Did you add tests to your changes? Did you modify tests to accommodate your changes?)
[//]: # (- Can this change be tested manually? How?)
Tested manually and created tests for the filters

## Screenshots of your changes (if appropriate):
[//]: # (It can also be a GIF to prove that your changes are working)
Screen Shot 2021-07-27 at 11 55 20

## Types of changes
What types of changes does your code introduce? Add an `x` in all the boxes that apply:
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Minor change (fix a typo, add a translation tag, add section to README, etc.)

### CODE REVIEW

**Comment by jsnshrmn** [comment]
Okay, I've manually tested emulating several devices (phone, tablet, desktop) and several languages (including hebrew, which is rtl) and it's functionally good IMO, other than the aforementioned issue with filtering languages. Moving on to a readthrough.

**Comment by suecarmol** [comment]
> The first thing I've noticed is that the language select filter has some odd behavior. On a fresh load/ hard reload (shift + F5) I find it almost impossible to select a language with a mouse click. Clicking left mouse also scrolls the list down and selects whichever language is beneath the cursor when the button is released. I'm able to (unreliably) select a language with the mouse once I've already fiddled with the control and there is some state.

I do not see this behavior in Chrome or Firefox for Mac. Do you have more details about how to reproduce this error?

**Comment by jsnshrmn** [comment]
It's very consistent for me on firefox 90 for ubuntu on this view, but I'm not seeing it on firefox 90 for windows or android on staging.

**Comment by jsnshrmn** [comment]
I will do some more testing to try to reproduce this in a clean  Linux vm to verify that it's not something weird about my system.

**Comment by jsnshrmn** [comment]
We verified that this was a firefox 90.0 bug that's resolved in the current release, 90.02.

File: TWLight/users/templates/users/filter_section.html [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+      data-target="#filterNavbar" aria-controls="filterNavbar" aria-expanded="false"
+      aria-label="Toggle navigation">
+    <span class="navbar-toggler-icon"></span>
+  </button>
+
+  <div id="filterNavbar" class="collapse navbar-collapse filter-container">
+    <div class="row filter-title-container">
+      <p class="filter-title">
+        {% comment %}Translators: On the MyLibrary page (https://wikipedialibrary.wmflabs.org/users/my_library), this label indicates the start of the filter section. {% endcomment %}
+        {% trans "Filters: " %}
```
Comment (jsnshrmn) [inline TWLight/users/templates/users/filter_section.html:13]: I think we should not be including the `: ` in our strings for translation. We had a couple of these sneak in over the last pr or two in `available_collection_tile.html` and `user_collection_tile.html` and we should bring these all into line with what we're doing elsewhere, which is like `{% trans "Filters" %}: `

**Review by jsnshrmn — CHANGES_REQUESTED** [review]
This looks great! I have one inline nitpick on localization strings, but other than that this is good to go IMO.

File: TWLight/users/templates/users/filter_section.html (hunk 1 above)
Comment (suecarmol) [reply TWLight/users/templates/users/filter_section.html]: Done!

**Comment by suecarmol** [comment]
I also fixed the `:` from the available collections and user collections templates

**Review by jsnshrmn — APPROVED** [review]

**Pull request merged**